exist-core/src/main/java/org/exist/xquery/value/FunctionParameterFunctionSequenceType.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery.value;
+
+import org.exist.xquery.Cardinality;
+import org.exist.xquery.ErrorCodes;
+import org.exist.xquery.XPathException;
+import org.exist.xquery.FunctionSignature;
+
+/**
+ * This class is used to specify the name and description of an XQuery function parameter of type function.
+ */
+public class FunctionParameterFunctionSequenceType extends FunctionParameterSequenceType {
+
+    private SequenceType[] parameters = null;
+    private SequenceType returnType = new SequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE);
+
+    /**
+     * mimics isOverloaded property of FunctionSignature
+     * @see org.exist.xquery.FunctionSignature
+     */
+    private int arity = -1;
+
+    /**
+     * @param attributeName     The name of the parameter in the <strong>FunctionSignature</strong>.
+     * @param primaryType       The <strong>Type</strong> of the parameter.
+     * @param parameterTypes    The <strong>parameters</strong> the function(s) must accept.
+     * @param returnType        The <strong>Type</strong> the function(s) needs to return.
+     * @param cardinality       The <strong>Cardinality</strong> of the parameter.
+     * @param description       A description of the parameter in the <strong>FunctionSignature</strong>.
+     * @see org.exist.xquery.FunctionSignature @see Type @see org.exist.xquery.Cardinality
+     */
+    public FunctionParameterFunctionSequenceType(final String attributeName, final int primaryType, final SequenceType[] parameterTypes, final SequenceType returnType, final Cardinality cardinality, final String description) {
```

Comment:
If I am not mistaken then the value supplied to `primaryType` appears to always be `Type.FUNCTION_REFERENCE`. If that is the case, then the parameter can be removed in favour of the constant.

In the same vein it might make sense to modify `FunctionParameterSequenceType` so that its public constructor cannot accept a `Type.FUNCTION_REFERENCE`; instead throwing an `IllegalArgumentException` indicating that `FunctionParameterFunctionSequenceType` should be constructed instead.